### ISSUE TITLE
Uncaught TypeError: Cannot read property 'left' of undefined with Chart.js 2.7

### ISSUE THREAD

**Issue by vyachekg** [chartjs/chartjs-plugin-annotation#87]
After updating chartjs to 2.7.0 get this error

>  Uncaught TypeError: Cannot read property 'left' of undefined.

```
min = helpers.isValid(options.xMin) ? options.xMin : xScale.getPixelForValue(chartArea.left);
max = helpers.isValid(options.xMax) ? options.xMax : xScale.getPixelForValue(chartArea.right);
```
How to reproduce.
When chart.update() is called and in options only one of xMin or xMax I get this error, because chartArea is undefined

**Comment by heyrex** [comment]
I have this bug too.

It seems that setDataLimits() is being called before the chartArea property has been created. setDataLimits() is called by initialize() inside module.exports() and that doesn't seem like the right place. I've scratched my head over this for an hour and cant see where the right place to call setDataLimits() is.

I did find that changing the code as follows made everything work ok for me on a bar chart:
```js
min = helpers.isValid(options.xMin) ? options.xMin : 0;
max = helpers.isValid(options.xMax) ? options.xMax : 1;
```
